app/models/skin.rb

```diff
@@ -276,12 +280,20 @@ def get_all_parents
   
   # This is the main function that actually returns code to be embedded in a page
   def get_style(roles_to_include = DEFAULT_ROLES_TO_INCLUDE)
-    style = ""
-    if self.get_role != "override" && self.get_role != "site"
-      style += AdminSetting.default_skin != Skin.default ? AdminSetting.default_skin.get_style(roles_to_include) : (Skin.get_current_site_skin ? Skin.get_current_site_skin.get_style(roles_to_include) : '')
+    Rails.cache.fetch('skins_style_tags/v1/' + self.get_role + '/' + \
+                                             self.id.to_s + '/' + \
+                                             AdminSetting.default_skin.updated_at.to_s + '/' + \
+                                             Skin.default.updated_at.to_s + '/' + \
+                                             Skin.get_current_version + '/' + \
+                                             (Rails.cache.fetch(skin_cache_key(self) || '0')
+                     ) do 
```

Comment:
The `do` on this line appear to be causing all the test failures. Also, when you update this, could you please take care of this comment? https://github.com/otwcode/otwarchive/pull/2227/files#r48453570 

I suspect merging with master might stop the unrelated prompt meme test changes from still showing in this diff, although I'm not 100% sure.